CAM-12752: add DB matrix stages

koevskinikola opened the pull request:
* Add stages for Engine/Engine Authorizations DB Unit tests;
* Add stages for Webapp/Webapp Authorizations DB Unit tests;
* Add missing stages in the second parallelization block;

Related to CAM-12752

yanavasileva commented on .ci/maven/nexus-settings.xml:
```diff
@@ -0,0 +1,141 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
```
Do we still need this file?

koevskinikola replied:
Removed it. I added it for testing purposes and forgot to remove it.

yanavasileva commented on Jenkinsfile:
```diff
@@ -351,46 +350,233 @@ pipeline {
+//        }
+        agent {
+          kubernetes {
+            yaml getAgent() + getDbAgent(env.DB)
```
Do we need the first agent?

koevskinikola replied:
I'm not sure about this. I'll check and let you know.

koevskinikola replied:
You're right, we don't need the first one! I'll also adjust it in the daily pipeline since I added the same thing there.

yanavasileva commented on Jenkinsfile:
```diff
@@ -351,46 +350,233 @@ pipeline {
           agent {
             kubernetes {
-              yaml getAgent('gcr.io/ci-30-162810/centos:v0.4.6', 16)
+              yaml getAgent()
```
We need more memory for this one otherwise, it will fail

yanavasileva commented on Jenkinsfile:
```diff
@@ -351,46 +350,233 @@ pipeline {
           agent {
             kubernetes {
-              yaml getAgent('gcr.io/ci-30-162810/centos:v0.4.6', 16)
+              yaml getAgent()
```
same here

yanavasileva commented on settings/maven/nexus-settings.xml:
```diff
@@ -2,26 +2,40 @@
       <mirrorOf>*</mirrorOf>
       <name>Camunda Nexus</name>
-      <url>https://app.camunda.com/nexus/content/groups/public</url>
+      <url>http://repository-ci-camunda-cloud.nexus:8081/content/groups/internal/</url>
```
We should not change this file

koevskinikola replied:
Reverted the changes. Made them during testing.

koevskinikola commented:
@yanavasileva, I implemented the review hints:

1. Removed and reverted the `nexus-settings.xml` files. 
2. Removed the extra `getAgent()` in the DB stages (also from the Daily pipeline).
3. Reverted the changes to the `getAgent()` lines where more resources are needed.
4. I also updated the `settings.xml` referrence to `camunda-maven-settings`  instead of `camunda-public-maven-settings`.

yanavasileva commented on Jenkinsfile:
```diff
@@ -434,3 +641,85 @@ void withLabels(String... labels) {
   }
 }
+
+void withDbLabels(String dbLabel) {
```
Is it a typo?
```suggestion
void withDbLabel(String dbLabel) {
```

koevskinikola replied:
Yes it is. Thanks!
